test-data/unit/check-classes.test

```diff
@@ -2770,9 +2770,9 @@ class B(A, metaclass=Y): pass  # E: Inconsistent metaclass structure for 'B'
 class M:
     x = 0  # type: int
 
-class A(metaclass=M): pass
+class A(metaclass=M): pass  # E: Metaclasses not inheriting from 'type' are not supported
 
-reveal_type(A.x)  # E: Revealed type is 'builtins.int'
+A.x  # E: "A" has no attribute "x"
```

Comment:
I would add a test that checks that "indirect" inheritance works, like:
```
class M1(type):
    x = 0
class M2(M1):
class C(metaclass=M2):
    pass
reveal_type(C.x) # E: Revealed type is 'builtins.int'
```
Otherwise LGTM.